GitPod Environment Not Set Up for Tests

When running an instance of the platform in GitPod, `npm run test:curriculum` throws this error:

```console
> cross-env FULL_OUTPUT=true mocha --delay --reporter progress

Error: Failed to launch the browser process!
/workspace/freeCodeCamp/curriculum/node_modules/puppeteer/.local-chromium/linux-800071/chrome-linux/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory


TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md

    at onClose (/workspace/freeCodeCamp/curriculum/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:193:20)
    at Interface.<anonymous> (/workspace/freeCodeCamp/curriculum/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:183:68)
    at Interface.emit (events.js:327:22)
    at Interface.close (readline.js:416:8)
    at Socket.onend (readline.js:194:10)
    at Socket.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1220:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @freecodecamp/curriculum@0.0.0-next.4 test:full-output: `cross-env FULL_OUTPUT=true mocha --delay --reporter progress`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @freecodecamp/curriculum@0.0.0-next.4 test:full-output script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gitpod/.npm/_logs/2020-09-29T19_02_52_374Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @freecodecamp/freecodecamp@0.0.1 test-curriculum-full-output: `cd ./curriculum && npm run test:full-output`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @freecodecamp/freecodecamp@0.0.1 test-curriculum-full-output script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gitpod/.npm/_logs/2020-09-29T19_02_52_393Z-debug.log
gitpod /workspace/freeCodeCamp $ 
```

Since the update of Puppeteer, I needed to run `sudo apt install libgbm1` to update, before the tests work. I assume something similar can be done for GitPod.

Useful resources courtesy of Mrugesh: 

Our config:
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/.gitpod.yml

Documentation:
https://www.gitpod.io/docs/config-docker/
https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#running-on-alpine

--- a/.gitpod.yml
+++ b/.gitpod.yml
@@ -1,5 +1,4 @@
-image:
-  file: .gitpod.Dockerfile
+image: gitpod/workspace-mongodb
 ports:
   - port: 27017 # mongodb
     onOpen: ignore

--- a/.gitpod.yml
+++ b/.gitpod.yml
@@ -1,60 +1,62 @@
-image: gitpod/workspace-mongodb
+image:
+  file: .gitpod.Dockerfile
 ports:
-- port: 27017 # mongodb
-  onOpen: ignore
-- port: 8000 # client
-  onOpen: open-preview
-- port: 9228 # node debug
-  onOpen: ignore
-- port: 3000 # api
-  onOpen: ignore
-- port: 9229 # node debug
-  onOpen: ignore
+  - port: 27017 # mongodb
+    onOpen: ignore
+  - port: 8000 # client
+    onOpen: open-preview
+  - port: 9228 # node debug
+    onOpen: ignore
+  - port: 3000 # api
+    onOpen: ignore
+  - port: 9229 # node debug
+    onOpen: ignore
 
 tasks:
-- before: |
-    echo '
-    export COOKIE_DOMAIN=gitpod.io
-    export HOME_LOCATION=$(gp url 8000)
-    export API_LOCATION=$(gp url 3000)
-    ' >> ~/.bashrc;
-    exit;
+  - before: |
+      echo '
+      export COOKIE_DOMAIN=gitpod.io
+      export HOME_LOCATION=$(gp url 8000)
+      export API_LOCATION=$(gp url 3000)
+      export CYPRESS_BASE_URL=$(gp url 8000)
+      ' >> ~/.bashrc;
+      exit;
 
-- name: db
-  # starting mongo in background, so it doesn't block prebuilds
-  before: >
-    mkdir -p /workspace/data &&
-    (mongod --dbpath /workspace/data &)
+  - name: db
+    # starting mongo in background, so it doesn't block prebuilds
+    before: >
+      mkdir -p /workspace/data &&
+      (mongod --dbpath /workspace/data &)
 
-- name: server
-  before: export COOKIE_DOMAIN=gitpod.io && export HOME_LOCATION=$(gp url 8000) && export API_LOCATION=$(gp url 3000)
-  # init is not executed for prebuilt workspaces and restarts,
-  # so we should put all the heavy initialization here
-  init: >
-    cp sample.env .env &&
-    npm ci &&
-    gp sync-done npm-ci &&
-    gp await-port 27017 &&
-    npm run seed &&
-    mongo --eval "db.fsyncLock(); db.fsyncUnlock()"
-  command: >
-    npm run ensure-env &&
-    gp await-port 27017 &&
-    npm run develop:server
+  - name: server
+    before: export COOKIE_DOMAIN=gitpod.io && export HOME_LOCATION=$(gp url 8000) && export API_LOCATION=$(gp url 3000) && export CYPRESS_BASE_URL=$(gp url 8000)
+    # init is not executed for prebuilt workspaces and restarts,
+    # so we should put all the heavy initialization here
+    init: >
+      cp sample.env .env &&
+      npm ci &&
+      gp sync-done npm-ci &&
+      gp await-port 27017 &&
+      npm run seed &&
+      mongo --eval "db.fsyncLock(); db.fsyncUnlock()"
+    command: >
+      npm run ensure-env &&
+      gp await-port 27017 &&
+      npm run develop:server
 
-- name: client
-  before: export HOME_LOCATION=$(gp url 8000) && export API_LOCATION=$(gp url 3000)
-  init: >
-    cd ./client &&
-    gp sync-await npm-ci &&
-    npm run prebuild &&
-    npm run build:workers &&
-    cd ..
-  command: >
-    gp await-port 3000 &&
-    cd ./client &&
-    npm run develop -- -H '0.0.0.0'
-  openMode: split-right
+  - name: client
+    before: export HOME_LOCATION=$(gp url 8000) && export API_LOCATION=$(gp url 3000) && export CYPRESS_BASE_URL=$(gp url 8000)
+    init: >
+      cd ./client &&
+      gp sync-await npm-ci &&
+      npm run prebuild &&
+      npm run build:workers &&
+      cd ..
+    command: >
+      gp await-port 3000 &&
+      cd ./client &&
+      npm run develop -- -H '0.0.0.0'
+    openMode: split-right
 
 github:
   prebuilds:

--- a/cypress-install.sh
+++ b/cypress-install.sh
@@ -1,0 +1,53 @@
+InstallCypress() {
+    sudo apt-get update && sudo apt-get install -y \
+        libgtk2.0-0 \
+        libgtk-3-0 \
+        libgbm-dev \
+        libnotify-dev \
+        libgconf-2-4 \
+        libnss3 \
+        libxss1 \
+        libasound2 \
+        libxtst6 \
+        xauth \
+        xvfb
+}
+
+InstallPuppeteer() {
+    sudo apt-get install -y \
+        ca-certificates \
+        fonts-liberation \
+        libappindicator3-1 \
+        libatk-bridge2.0-0 \
+        libatk1.0-0 \
+        libc6 \
+        libcairo2 \
+        libcups2 \
+        libdbus-1-3 \
+        libexpat1 \
+        libfontconfig1 \
+        libgbm1 \
+        libgcc1 \
+        libglib2.0-0 \
+        libnspr4 \
+        libpango-1.0-0 \
+        libpangocairo-1.0-0 \
+        libstdc++6 \
+        libx11-6 \
+        libx11-xcb1 \
+        libxcb1 \
+        libxcomposite1 \
+        libxcursor1 \
+        libxdamage1 \
+        libxext6 \
+        libxfixes3 \
+        libxi6 \
+        libxrandr2 \
+        libxrender1 \
+        lsb-release \
+        wget \
+        xdg-utils
+}
+
+InstallCypress
+InstallPuppeteer

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
     "precypress": "node ./cypress-install.js",
     "cypress": "cypress",
     "cypress:install": "cypress install && echo 'for use with ./cypress-install.js'",
+    "cypress:install-build-tools": "sh ./cypress-install.sh",
     "cypress:dev:run": "npm run cypress -- run",
     "cypress:dev:watch": "npm run cypress -- open",
     "cypress:prd:run": "npm run cypress -- run",

